### PR TITLE
Add validation and model checks

### PR DESCRIPTION
**Pull request by Tayyab-Kamboh**
Following tasks were completed: 

- Use validations for models.
-  Write unit tests for models.

### CODE REVIEW

File: spec/models/post_spec.rb [hunk 1]
```diff
@@ -0,0 +1,85 @@
+require 'rails_helper'
+
+RSpec.describe Post, type: :model do
+  describe 'associations' do
```
Comment (Lizdev-05) [inline spec/models/post_spec.rb:4]: - [x] Please make sure that you add a test that a post title must not exceed 250 characters. :pray:

**Review by Lizdev-05 — CHANGES_REQUESTED** [review]
Hi @Tayyab-Kamboh , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
:heavy_check_mark: Professional Readme file
:heavy_check_mark: All your tests passes :+1: 
:heavy_check_mark: No linter error

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/post_spec.rb (hunk 1 above)
Comment (Tayyab-Kamboh) [reply spec/models/post_spec.rb:4]: @Lizdev-05  Donee✌️

**Review by Lizdev-05 — APPROVED** [review]
Hi @Tayyab-Kamboh  

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
 :heavy_check_mark: Professional Readme file
:heavy_check_mark: All your tests pass +1
:heavy_check_mark: No linter error

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**